packages/next-auth/src/core/pages/signin.tsx

```diff
@@ -71,7 +71,9 @@ export default function SigninPage(props: SignInServerPageParams) {
     default: "Unable to sign in.",
   }
 
-  const error = errorType && (errors[errorType] ?? errors.default)
+  const error =
+    (errorType && Object.hasOwn(errors, errorType) && errors[errorType]) ||
+    errors.default
```

Comment:
this will actually always return `error.default` even when `errorType === undefined`.
My suggestion is:
```suggestion
    const error = errorType && ((Object.hasOwn(errors, errorType)) ? errors[errorType] : errors.default)
```

Tested locally from my side 👀

Reply:
👌 Didn't know the `undefined` was intended!